Makefile.in

```diff
@@ -70,6 +71,16 @@ DEPENDENCY_CFLAG = @DEPENDENCY_CFLAG@
 	@rm -f $@
 	$(CC) $(FULL_CFLAGS) -c $(srcdir)/$*.c
 
+# Teaching plain make what to do with .rc files...
+# Simply saying '.rc.o:' only works if .rc files would be part of make's
+# known suffixes.
+# This could be done with '.SUFFIXES: .rc' but that would mean adding yet
+# another line (he thought while typing a five-line long comment).
+
+%.o : %.rc
+	$(WINDRES) -DPACKAGE_VERSION_WIN="`egrep -o '([0-9]{1,}\.)+[0-9]{1,}' \
+        $(srcdir)/VERSION | tr '.' ','`,0" $^ -o $@
```

Comment:
Maybe this could be done "*simpler*" if one assumes GNU-make. E.g.:
```
comma := ,
VERSION = $(shell cat ./VERSION)
RC_VERSION := $(VERSION:-PRE-GIT=)
RC_VERSION := $(subst .,$(comma),$(RC_VERSION)),0

$(WINDRES) -DPACKAGE_VERSION_WIN='$(RC_VERSION)' ...
```
Or maybe just include a generate `version.c` into both `tcpdump-rc.rc` and `tcpdump.c`. Similar to what libpcap does
`with version.c.in`

PS. The `tcpdump.ico` was nice. I've added it to my builds of both tcpdump and libpcap.

Reply:
Glad you like the icon. Here is the layered PSD in case you want to tweak it.
[icon.zip](https://github.com/the-tcpdump-group/tcpdump/files/1533302/icon.zip)

Reply:
Maybe this could be done "simpler" if one assumes GNU-make. E.g.:

This is simpler. Even if only simpler to look at... 😄 
egrep is already a dependency for building tcpdump with the autotools.
I also prefer GNU-make, but this, and the fact that it took me forever to get `sh` + `make` to parse this properly (automake, where are you? 😢 ), are my main arguments for wanting to stick with this.

Reply:
> Or maybe just include a generated version.c into both tcpdump-rc.rc and tcpdump.c. Similar to what libpcap does with version.c.in.

I get the first part, but I wouldn't know how tcpdump's version could be converted to PACKAGE_VERSION_WIN (with commas instead of dots etc) this way. Thoughts?

Reply:
>  (with commas instead of dots etc) this way. Thoughts?

I use some [home-brewed](https://gist.github.com/gvanem/a6eaf29bda43b82fdd8556ed7f6efe2a) GNU-makefiles for MinGW __and__ MSVC. Since both *autotools* and *Nmake* suck so hard. I convert from `.` to `,` using GnuMake's ` $(subst)`  function:
```
comma := ,
_RC_VERSION := $(subst .,$(comma),$(_RC_VERSION)),0
```
and use that in the *generated* `version.c` which gets included into `windump.rc`. A bit complex, but it does what I want. Namely showing me which compiler built that program. This is always a question when things doesn't work; like mixing .DLLs and .EXEs built with both MinGW and MSVC. I'm not sure this is 100% safe with WinPcap/wpcap.

Reply:
The makefile is very nice. Great job. ascii.exe? Get outta here!
I am interested to know how PCAP_TRACE works. Maybe you can elaborate.
Also do all tests pass? I remember mine struggling with some of them. Didn't try with dos2unix though. Thought that's unneeded because gitattibute...

A generated version.c which gets included into windump.rc actually has some benefits. It's nice to know who/what build it. It's also good if you don't use the autotools build system. But you would still need to edit the top-level makefile to generate a version.c that look like yours and add the rc file to tcpdump's sources for those who do.
 
The generation of a version.c with your proposed content could be also easily generated with a VS prebuild event (aka a batch/powershell script inside a project file). That's actually the plan once I finish pushing this wagon over the hill.